http/handler.go

```diff
@@ -577,6 +577,34 @@ func request(core *vault.Core, w http.ResponseWriter, rawReq *http.Request, r *l
 		return nil, false, true
 	}
 
+	if resp != nil && len(resp.Headers) > 0 {
+		// Set this here so it will take effect regardless of any other type of
+		// response processing
+		if len(resp.Headers) > 0 {
```

Comment:
This length check is redundant with the one above.

Reply:
Thanks, fixed!